ast/parser.go

```diff
@@ -2373,6 +2373,21 @@ func (b *metadataParser) Parse() (*Annotations, error) {
 	}
 
 	result.Location = b.loc
+
+	// recreate original text of entire metadata block for location text attribute
+	sb := strings.Builder{}
+	sb.WriteString("# METADATA\n")
+
+	lines := bytes.Split(b.buf.Bytes(), []byte{'\n'})
+
+	for _, line := range lines[:len(lines)-1] {
+		sb.WriteString("# ")
+		sb.Write(line)
+		sb.WriteByte('\n')
+	}
```

Comment:
Is this not the same as replacing every "\n" with "\n# "? like, [`strings.ReplaceAll`](https://pkg.go.dev/strings#ReplaceAll)?

Reply:
Yeah, I hadn't considered that. Will do for next time, but feels like both are OK in this location.

Reply:
Also care would need to be applied for the first line, as it'll need a `# '  too, but isn't preceeded by a newline.

Reply:
This works, but while it's likely faster, I'm not so sure about readability:
```go
	block := []byte("# METADATA\n# ")
	block = append(block, bytes.TrimSuffix(
		bytes.ReplaceAll(b.buf.Bytes(), []byte{'\n'}, []byte{'\n', '#', ' '}),
		[]byte{'\n', '#', ' '})...,
	)

	result.Location.Text = block
```
which one do you prefer?